cmd/admission-controller/main.go

```diff
@@ -68,7 +81,66 @@ func main() {
 		glog.Fatalf("failed to get kubernetes Clientset: %v", err)
 	}
 
-	webhookServer := webhook.NewWebHookServer(kubeCli, cli, certFile, keyFile)
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeCli, controller.ResyncDuration)
+	for v, synced := range kubeInformerFactory.WaitForCacheSync(wait.NeverStop) {
+		if !synced {
+			glog.Fatalf("error syncing informer for %v", v)
+		}
+	}
+
+	webhookServer := webhook.NewWebHookServer(kubeCli, cli, kubeInformerFactory, certFile, keyFile)
+	controllerCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	kubeInformerFactory.Start(controllerCtx.Done())
+	for v, synced := range kubeInformerFactory.WaitForCacheSync(wait.NeverStop) {
+		if !synced {
+			glog.Fatalf("error syncing informer for %v", v)
+		}
+	}
+	glog.Infof("cache of informer factories sync successfully")
+
+	if err := webhookServer.Run(); err != nil {
+		glog.Errorf("stop http server %v", err)
+	}
+
+	/*
+	* currently we only use one replica for admission-controller-webhook
+	 */
+
+	//rl := resourcelock.EndpointsLock{
+	//	EndpointsMeta: metav1.ObjectMeta{
+	//		Namespace: ns,
+	//		Name:      "admission-controller-webhook",
+	//	},
+	//	Client: kubeCli.CoreV1(),
+	//	LockConfig: resourcelock.ResourceLockConfig{
+	//		Identity:      hostName,
+	//		EventRecorder: &record.FakeRecorder{},
+	//	},
+	//}
+	//onStarted := func(ctx context.Context) {
+	//	if err := webhookServer.Run(); err != nil {
+	//		glog.Errorf("stop http server %v", err)
+	//	}
+	//}
+	//
+	//onStopped := func() {
+	//	glog.Fatalf("leader election lost")
+	//}
+
+	//go wait.Forever(func() {
+	//	leaderelection.RunOrDie(controllerCtx, leaderelection.LeaderElectionConfig{
+	//		Lock:          &rl,
+	//		LeaseDuration: leaseDuration,
+	//		RenewDeadline: renewDuration,
+	//		RetryPeriod:   retryPeriod,
+	//		Callbacks: leaderelection.LeaderCallbacks{
+	//			OnStartedLeading: onStarted,
+	//			OnStoppedLeading: onStopped,
+	//		},
+	//	})
+	//}, waitDuration)
```

Comment:
what's the problem of leader election? it's ok to use leader election with 1 replica.

Reply:
fix